### PR TITLE
Bug 1322768 - Use Spark to list file summaries

### PR DESCRIPTION
**Pull request by maurodoglio**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/python_moztelemetry/108)

### CODE REVIEW

File: moztelemetry/dataset.py [hunk 1]
```diff
@@ -156,10 +156,8 @@ def summaries(self, limit=None):
 
         with futures.ProcessPoolExecutor(MAX_CONCURRENCY) as executor:
             scanned = self._scan(schema, [self.prefix], clauses, executor)
```
Comment (vitillo) [inline moztelemetry/dataset.py]: How much of summaries is spent in scanning the directories vs listing the keys usually?

Comment (maurodoglio) [inline moztelemetry/dataset.py]: I don't remember the numbers, but in most of the cases the long pole is listing the keys. It really depends on the filter though, the extreme cases (i.e. either no filters or only one filter on buildid) behave very differently.

Comment (maurodoglio) [inline moztelemetry/dataset.py]: Btw before we merge this I want to run a benchmark to verify some numbers.

Comment (maurodoglio) [inline moztelemetry/dataset.py]: I timed the execution of the first Dataset query in the hello world notebook and I can confirm the long pole is listing the files; 80% of the time is spent on it.
Unsurprisingly the spark implementation is slightly slower in some cases and faster in other cases. Basically when there are a lots of directories to list on, the spark implementation is faster on a multi node cluster. For example with the following query on a 10 nodes cluster, the execution time went down from 30s to 20s:
```python
pings = Dataset.from_source("telemetry") \
    .where(docType='main') \
    .where(submissionDate=lambda x: "20161201" < x < "20170201") \
    .where(appUpdateChannel="nightly") \
    .records(sc, sample=0.1)
```
On the other hand, some queries like
```python
pings = Dataset.from_source("telemetry") \
    .where(docType='main') \
    .where(submissionDate="20161101") \
    .where(appUpdateChannel="nightly") \
    .records(sc, sample=0.1)
``` 
take about 3 seconds more. Overall, I think having a progress bar is worth a small speed penalty you get in some cases.

Comment (vitillo) [inline moztelemetry/dataset.py]: r+

**Pull request merged**